pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -272,16 +273,59 @@ def _on_evaluation_epoch_end(self) -> None:
         self.trainer.logger_connector.on_epoch_end()
 
     def _print_results(self, results: List[_OUT_DICT], stage: RunningStage) -> None:
-        # TODO: this could be updated to look nicer
-        from pprint import pprint
-
-        print("-" * 80)
-        for i, metrics_dict in enumerate(results):
-            print(f"DATALOADER:{i} {stage.upper()} RESULTS")
-            pprint(
-                {
-                    k: (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
-                    for k, v in metrics_dict.items()
-                }
-            )
-            print("-" * 80)
+        if _RICH_AVAILABLE:
+            from rich.console import Console
+            from rich.table import Table
+
+            console = Console()
+            table = Table()
+
+            table.add_column("Metric", justify="center", style="cyan")
+
+            new_results = []
+
+            for result in results:
+                new_keys = [i.split("/dataloader_idx_")[0] for i in result.keys()]
+                new_results.append(dict(zip(new_keys, result.values())))
+
+            unique_keys = sorted(set().union(*(d.keys() for d in new_results)))
+
+            rows = [[i] for i in unique_keys]
+
+            for i, metrics_dict in enumerate(new_results):
+                table.add_column(f"DATALOADER {i}", justify="center", style="magenta")
+
+                for j, metric in enumerate(rows):
+                    v = metrics_dict.get(metric[0])
+                    v = (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
+                    rows[j].append(f"{v}")
+
+            for row in rows:
+                table.add_row(*row)
+
+            console.print("", table)
+
+        else:
+            new_results = []
+
+            for result in results:
+                new_keys = [i.split("/dataloader_idx_")[0] for i in result.keys()]
+                new_results.append(dict(zip(new_keys, result.values())))
+
+            unique_keys = sorted(set().union(*(d.keys() for d in new_results)))
+
+            rows = [[] for x in range(len(unique_keys))]
+
+            for i, metrics_dict in enumerate(new_results):
+                for j in range(len(rows)):
+                    v = metrics_dict.get(unique_keys[j])
+                    v = (v.item() if v.numel() == 1 else v.tolist()) if isinstance(v, torch.Tensor) else v
+                    rows[j].append(f"{v}")
+
+            cols = [f"DATALOADER {i}" for i in range(len(results))]
+
+            row_format = "{:^25}" * (len(cols) + 1)
```

Comment:
Do you think we could improve this to support long metric names?

Current output is:
```python
                               DATALOADER 0             DATALOADER 1                                                                                                                                                                                                                                                                                                               
a really really really long metric name   0.20651429891586304      0.4664147198200226    
           foo              0.10347846895456314      0.13967294991016388   
       valid_loss           -0.1181236281991005       2.235642671585083    
```

Reply:
I've made the column width adaptable in the native version (in rich table this is done automatically) with a minimum width of 25.